akka-testkit/src/test/scala/akka/testkit/TestProbeSpec.scala

```diff
@@ -39,6 +41,38 @@ class TestProbeSpec extends AkkaSpec with DefaultTimeout {
       probe1.expectMsg(0 millis, "some hint here", "world")
     }
 
+    "create a child when invoking actorOf" in {
+      val probe = TestProbe()
+      val child = probe.childActorOf(TestActors.echoActorProps)
+      child.path.parent should be(probe.ref.path)
+
+      val namedChild = probe.childActorOf(TestActors.echoActorProps, name = Some("actorName"))
+      namedChild.path.name should be("actorName")
+    }
+
+    "restart a failing child if the given supervisor says so" in {
+      val restarts = new AtomicInteger(0)
+
+      class FailingActor extends Actor {
+        override def receive = msg ⇒ msg match {
+          case _ ⇒
+            throw new RuntimeException("simulated failure")
+        }
+
+        override def postRestart(reason: Throwable): Unit = {
+          restarts.incrementAndGet()
+        }
+      }
+
+      val probe = TestProbe()
+      val child = probe.childActorOf(Props(new FailingActor), supervisorStrategy = Some(SupervisorStrategy.defaultStrategy))
+
+      eventually {
```

Comment:
I'd prefer our own `awaitAssert`. It probably does the same thing, but then we are using the same timeout handling as elsewhere.